src/pages/ScholarshipForm.jsx

```diff
@@ -42,26 +133,54 @@ function ScholarshipForm() {
     <Container maxWidth="md">
       <form>
         <h1>Submit a Scholarship</h1>
-        {requiredTextField('name', 'Scholarship Name')}
+        <div>
+          {requiredTextField(
+            'name',
+            'Scholarship Name',
+            formFieldErrors.name,
+            'Please enter a name'
+          )}
+        </div>
         <DatePicker
           id="deadline"
           label="Deadline"
+          error={formFieldErrors.deadline}
+          helperText="Please enter a valid date"
           value={formFieldStates.deadline}
           onChange={(date) =>
             setFormFieldStates({ ...formFieldStates, deadline: date })
           }
         />
-        {requiredTextField('description', 'Description')}
-        {requiredTextField('website', 'Website')}
+        <div>
+          {requiredTextField(
+            'description',
+            'Description',
+            formFieldErrors.description,
+            'Please enter a description',
+            true
+          )}
```

Comment:
With so many parameters it gets hard to read this and understand values like `true` by themselves without referencing the function component. I suggest capitalizing the component name (and maybe putting in its own file) and accepting properties so you can use it like this:
```jsx
<RequiredTextField id="description" label="Description" ... multiline/>
```

Also, it seems like `id` or `label` could be calculated from each other (at least in these cases). That could potentially be an optimization to make but I can also see how it could make things difficult if that's ever not the case.